services/item-service.js

```diff
@@ -125,13 +128,156 @@ exports.getPants = async function (itemId) {
  */
 exports.getAllItems = async function () {
     try {
-        let items = await Item.findAll();
+        let items = await Item.findAll({
+            include: [{
+                model: Shirts, as: "shirts"
+            }, {
+                model: Shoes, as: "shoes"
+            },
+            {
+                model: Pants, as: "pants"
+            },
+            {
+                model: Suits, as: "suits"
+            }]
+        });
+        console.log(items)
+        return items;
+    } catch (e) {
+        throw new InternalErrorException("A problem occurred when retrieving items", e);
+    }
+}
+
+
+/**
+ * Retrieves and returns all items from the Items table of a specific type and gender
+ * @param {string} gender 
+ * @param {enum} type
+ * 
+ */
+
+exports.getItemsByCategory = async function (gender, type) {
+    try {
+        let items = null
+        switch (type) {
+            case "shoes":
+                items = await Item.findAll({
+                    where: { gender: gender, type: type },
+                    include: [{
+                        model: Shoes, as: "shoes"
+                    }]
+                });
+                break;
+
+            case "suits":
+                items = await Item.findAll({
+                    where: { gender: gender, type: type },
+                    include: [{
+                        model: Suits, as: "suits"
+                    }]
+                });
+                break;
+
+            case "pants":
+                items = await Item.findAll({
+                    where: { gender: gender, type: type },
+                    include: [{
+                        model: Pants, as: "pants"
+                    }]
+                });
+                break;
+
+
+            case "shirts":
+                items = await Item.findAll({
+                    where: { gender: gender, type: type },
+                    include: [{
+                        model: Shirts, as: "shirts"
+                    }]
+                });
+                break
+        }
+        return items;
+    } catch (e) {
+        throw new InternalErrorException("A problem occurred when retrieving items", e);
+    }
+}
+
```

Comment:
lots of space between functions...this is just my preference but if you want this to stay then idc